vlc-android/src/org/videolan/vlc/gui/network/MRLPanelFragment.kt

```diff
@@ -0,0 +1,135 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ */
+package org.videolan.vlc.gui.network
+
+import android.net.Uri
+import android.os.Bundle
+import android.support.design.widget.TextInputLayout
+import android.support.v4.app.DialogFragment
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.text.TextUtils
+import android.view.KeyEvent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
+import android.widget.TextView
+
+import org.videolan.medialibrary.media.MediaWrapper
+import org.videolan.vlc.R
+import org.videolan.vlc.VLCApplication
+import org.videolan.vlc.gui.DialogActivity
+import org.videolan.vlc.gui.helpers.UiTools
+import org.videolan.vlc.media.MediaUtils
+import org.videolan.vlc.util.Constants.KEY_MRL
+
+const val TAG = "VLC/MrlPanelFragment"
+
+class MRLPanelFragment : DialogFragment(), View.OnKeyListener, TextView.OnEditorActionListener, View.OnClickListener, MRLAdapter.MediaPlayerController {
+    private lateinit var adapter: MRLAdapter
+    private lateinit var editText: TextInputLayout
+
+    val isEmpty: Boolean
+        get() = adapter.isEmpty
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(DialogFragment.STYLE_NO_FRAME, 0)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val v = inflater.inflate(R.layout.mrl_panel, container, false)
+        editText = v.findViewById(R.id.mrl_edit)
+        editText.editText?.setOnKeyListener(this)
+        editText.editText?.setOnEditorActionListener(this)
+        val recyclerView = v.findViewById<RecyclerView>(R.id.mrl_list)
+        recyclerView.addItemDecoration(DividerItemDecoration(activity, DividerItemDecoration.VERTICAL))
+        recyclerView.layoutManager = LinearLayoutManager(activity)
+        adapter = MRLAdapter(this)
+        recyclerView.adapter = adapter
+        v.findViewById<View>(R.id.send).setOnClickListener(this)
+        return v
+    }
+
+    override fun onStart() {
+        super.onStart()
+        updateHistory()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(KEY_MRL, editText.editText?.text.toString())
+    }
+
+    override fun onViewStateRestored(savedInstanceState: Bundle?) {
+        super.onViewStateRestored(savedInstanceState)
+        val mrl = savedInstanceState?.getString(KEY_MRL)
```

Comment:
You can combine thses 2 lines:
```
val mrl = savedInstanceState?.getString(KEY_MRL)?.apply {
    editText.editText?.setText(this)
}

Reply:
I removed onViewStateRestored in the next commits. because of ViewModel and two-way data binding we don't need this anymore.